File: interfaces/src/main/java/com/nerdscorner/mvplib/interfaces/model/BaseModel.java
package com.nerdscorner.mvplib.interfaces.model;

import com.nerdscorner.mvplib.interfaces.presenter.BasePresenter;

public class BaseModel<P extends BasePresenter> {

    protected P presenter;

    public void setPresenter(P presenter) {
        this.presenter = presenter;
    }

    public void onResume() {
    }

    public void onPause() {
    }

    public void onStart() {
    }

    public void onStop() {
    }
}

File: commons/src/main/java/com/nerdscorner/mvplib/commons/mvp/model/BaseModel.kt
package com.nerdscorner.mvplib.commons.mvp.model

open class BaseModel {

    open fun onResume() {}

    open fun onPause() {}

    open fun onStart() {}

    open fun onStop() {}

    open fun onDestroy() {}

    open fun onDestroyView() {}
}